pkg/plugin/plugin.go

```diff
@@ -437,3 +456,40 @@ func (p *PowerPlugin) monitorSocketHealth() {
 		}
 	}
 }
+
+// Read config map file
+func loadDevicePluginConfig() (*api.DevicePluginConfig, error) {
+	klog.Infof("Attempting to read config file from: %s", configPath)
+
+	info, err := os.Stat(filepath.Clean(configPath))
+	if err != nil {
+		if os.IsNotExist(err) {
+			klog.Warningf("Config file not found at %s. Proceeding with default configuration.", configPath)
+			return &api.DevicePluginConfig{}, nil
+		}
+		klog.Warningf("Unable to stat config file: %v", err)
+		return nil, err
+	}
+
+	if info.IsDir() {
+		klog.Warningf("Config path %s is a directory, not a file. Proceeding with default configuration.", configPath)
+		return &api.DevicePluginConfig{}, nil
+	}
+
+	data, err := os.ReadFile(filepath.Clean(configPath))
+	if err != nil {
+		klog.Warningf("Unable to read config file: %v", err)
+		return nil, err
+	}
+
+	klog.Infof("Raw config data: %s", string(data))
+
+	var config api.DevicePluginConfig
+	if err := json.Unmarshal(data, &config); err != nil {
+		klog.Errorf("Failed to unmarshal config file: %v", err)
+		return nil, err
+	}
+
+	klog.Infof("Config loaded successfully: %+v", config)
```

Comment:
```suggestion
	klog.Infof("Config loaded successfully")
```
It is a configmap and shouldn't have sensitive data in it. As a practice, let's not log the object out. If it was a secret, it'd present other problems of potentially disclosing username/passwords in logs.